lib/spack/spack/test/concretize_preferences.py

```diff
@@ -114,11 +114,50 @@ def test_preferred_providers(self):
         spec = concretize('mpileaks')
         assert 'zmpi' in spec
 
+    def test_preferred(self):
+        """"Test packages with some version marked as preferred=True"""
+        spec = Spec('preferred-test')
+        spec.concretize()
+        assert spec.version == spack.spec.Version('0.2.15')
+
+        # now add packages.yaml with versions other than preferred
+        conf = syaml.load("""\
+preferred-test:
+        version: [0.2.16]
+""")
+        spack.config.update_config('packages', conf, 'concretize')
```

Comment:
something changed in Spack after rebase and now I cant make unit test pick up the `yaml` file. I confirm that this is indeed the reasons for failing tests by printing
```
        for v in usable:
            print (v)
            print (yaml_prefs(v))
```
right after this line https://github.com/LLNL/spack/blob/develop/lib/spack/spack/concretize.py#L181
It gives me:
```
0.2.15
0
develop
0
0.2.15
0
0.2.15.develop
0
0.2.15
0
develop
0
```

In the last version of this PR, the issue was magically sorted out by `spack.package_prefs._pkgsort = PreferredPackages()`.

@tgamblin @alalazo do  you happen to know why `yaml` are not being red correctly here?